mmf/trainers/core/evaluation_loop.py

```diff
@@ -17,33 +17,53 @@
 
 class TrainerEvaluationLoopMixin(ABC):
     def evaluation_loop(
-        self, loader, use_tqdm: bool = False, single_batch: bool = False
+        self, dataset_type: str, use_tqdm: bool = False, single_batch: bool = False
     ) -> Tuple[Dict[str, Any], Type[Meter]]:
         meter = Meter()
+        reporter = self.dataset_loader.get_test_reporter(dataset_type)
```

Comment:
Since, after this change in `DatasetLoader`, val_dataset and test_dataset are essentially useless, I would suggest that we move this to `DatasetLoader` and instead of using `MulitDatasetLoader` to initialize val and test dataloader use this. Also, now, I think it is right time to move `test_reporter.py` to be under datasets folder as well.

Reply:
After further inspection, I think it would be better to keep the `MulitDatasetLoader` for val and test datasets, for the following reasons:

1) val_dataset and test_dataset are still used in this case -- currently, `TestReporter` dependents on `MulitDatasetLoader` to load each evaluation dataset. During execution, it accesses each test dataset through `MulitDatasetLoader`'s `get_datasets` function.
2) It would be better to keep just one, same mechanism to load multiple datasets rather than maintaining a separate multi-dataset loading mechanism, so using those that already exists in `MulitDatasetLoader` probably works better.
3) Keeping `MulitDatasetLoader` for val and test datasets would lead to a smaller refactorization and more backward compatibility in the MMFTrainer (so that if people subclass this trainer and uses val datasets, it won't be broken).

Reply:
I like your suggestion, but maintainability wise it is an extra burden. I will fix this in my change around datamodules.